R/zzz.R

```diff
@@ -44,6 +44,44 @@ default_linters <- modify_defaults(
 #' There is a list for the default elements and another that contains all available elements.
 #' Use [modify_defaults()] to produce a custom list.
 #'
+#' @details
+#' The following functions can be regarded as undesirable:
+#'
+#'  * `attach()` modifies the global search path. Use roxygen2's @importFrom statement in packages, or `::` in scripts.
+#'  * `browser()` pauses execution when run and is likely a leftover from debugging. It should be removed.
+#'  * `debug()` traps a function and causes execution to pause when that function is run. It should be removed.
+#'  * `debugcall()` works similarly to `debug()`, causing execution to pause. It should be removed.
+#'  * `debugonce()` is only useful for interactive debugging. It should be removed.
+#'  * `detach()` modifies the global search path. Detaching environments from the search path should not be necessary
+#'    in production code.
+#'  * `ifelse()` isn't type stable. Use an `if`/`else` block for scalar logic, or use
+#'    `dplyr::if_else()`/`data.table::fifelse()` for type stable vectorized logic.
+#'  * `.libPaths()` permanently modifies the library location. Use `withr::with_libpaths()` for a temporary change
+#'    instead.
+#'  * `library()` modifies the global search path. Use roxygen2's @importFrom statement in packages, or `::` in scripts.
+#'  * `loadNamespace()` doesn't provide an easy way to signal failures. Use the return value of `requireNamespace()`
+#'    instead.
+#'  * `mapply()` isn't type stable. Use `Map()` to guarantee a `list()` is returned and simplify accordingly.
+#'  * `options()` permanently modifies the session options. Use `withr::with_options()` for a temporary change instead.
+#'  * `par()` permanently modifies the graphics device parameters. Use `withr::with_par()` for a temporary change
+#'    instead.
+#'  * `require()` modifies the global search path. Use roxygen2's @importFrom statement in packages, or `::` in scripts.
```

Comment:
slightly odd for `library()` and `require()` to have the same diagnosis -- isn't part of this to steer towards `library()` in scripts?

Reply:
That's true. For scripts, `library()` shouldn't be undesirable and `require()` should be avoided.